apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SqlResultStream.cs

```diff
@@ -111,7 +111,12 @@ public async Task<bool> MoveNext(CancellationToken cancellationToken)
         private async Task<PartialResultSet> ComputeNextAsync(CancellationToken cancellationToken)
         {
             // The retry state is local to the method as we're not trying to handle callers retrying.
-            RetryState retryState = new RetryState(_client.Settings.Scheduler ?? SystemScheduler.Instance, _retrySettings);
+            // Consecutive retryable errors will be retried until the call succeeds, or until the call times out or is cancelled.
+            RetryState retryState = new RetryState(_client.Settings.Scheduler ?? SystemScheduler.Instance, _retrySettings, maxConsecutiveErrors: int.MaxValue);
+            var clock = _client.Settings.Clock ?? SystemClock.Instance;
+            var startTime = clock.GetCurrentDateTimeUtc();
+            var originalTimeout = _callSettings.Expiration?.Timeout ?? _client.Settings.ExecuteStreamingSqlSettings.Expiration?.Timeout ?? TimeSpan.FromMilliseconds(3600000);
```

Comment:
I think that if the Expiration is a deadline, it should be respected as well.

Reply:
I've updated the implementation to respect a deadline as well. There is however one potential issue with using a deadline here:
1. The `CallSettings` are passed in to constructor of `SqlResultStream`, for example when [ExecuteQueryAsync](https://github.com/googleapis/google-cloud-dotnet/blob/34b71e8b87efdbc3043906d9b2dd7d1b8e9bb43d/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/EphemeralTransaction.cs#L141) is called by a client application.
2. That means that the same `CallSettings` are used for all calls to `MoveNext()`, which means that a deadline is applied to the entire stream and not to a single RPC.

Currently it's not a real-life problem as a user can't pass in custom `CallSettings` and the default `CallSettings` currently use a timeout. Each timeout will only apply to a single RPC invocation.

Reply:
> That means that the same CallSettings are used for all calls to MoveNext(), which means that a deadline is applied to the entire stream and not to a single RPC

I might be missing something here, but I don't see how this is a problem. A deadline is a deadline, and applying it to the whole operation means that it is effectively applied to each single RPC. There's no way that the whole operation can finish before the deadline if the single RPCs didn't all finish before the deadline?